autogluon/task/base/base_task.py

```diff
@@ -17,8 +21,18 @@
     'random': FIFOScheduler,
     'skopt': FIFOScheduler,
     'hyperband': HyperbandScheduler,
-    'rl': RLScheduler,
-}
+    'rl': RLScheduler}
+
+
+def create_scheduler(train_fn, search_strategy, scheduler_options):
+    if isinstance(search_strategy, str):
+        scheduler_cls = schedulers[search_strategy.lower()]
+    else:
+        assert callable(search_strategy)
+        scheduler_cls = search_strategy
+        scheduler_options = copy.copy(scheduler_options)
+        scheduler_options['searcher'] = 'random'
```

Comment:
why is this always set to 'random'?

Reply:
This was like that in the current code, I did not change it.
I do not know who would call 'fit' with search_strategy being not a str (then, it must be a scheduler class).

We could remove this altogether, because passing a non-str as search_strategy would fail before in all these cases (and did so in the old code). Shall I throw this out, and insist on search_strategy being a str? The comment says so anyway.